Fix infinite loop for token format

rockerBOO opened the pull request:
I am setting up ClientCapabilties like

```go
		semanticTokens := protocol.SemanticTokensClientCapabilities{
			TokenTypes: []string{
				"namespace", "type", "class", "enum", "interface", "struct",
				"typeParameter", "parameter", "variable", "property", "enumMember",
				"event", "function", "method", "macro", "keyword", "modifier",
				"comment", "string", "number", "regexp", "operator",
			},

			TokenModifiers: []string{
				"declaration", "definition", "readonly", "static", "deprecated",
				"abstract", "async", "modification", "documentation", "defaultLibrary",
			},
			Formats:               []protocol.TokenFormat{protocol.TokenFormatRelative},
			DynamicRegistration:   false,
			MultilineTokenSupport: false,
			ServerCancelSupport:   false,
			AugmentsSyntaxTokens:  true,
		}

		params := protocol.InitializeParams{
			ProcessId: &process_id,
			ClientInfo: &protocol.ClientInfo{
				Name:    "MCP-LSP Bridge",
				Version: "1.0.0",
			},
			RootUri:          &root_uri,
			WorkspaceFolders: &workspaceFolders,
			Capabilities: protocol.ClientCapabilities{
				TextDocument: &protocol.TextDocumentClientCapabilities{
					SignatureHelp:  &protocol.SignatureHelpClientCapabilities{},
					SemanticTokens: &semanticTokens,
				},
			},
		}
```

but it causes it to loop infinitely when converting to JSON. Due to the protocol.TokenFormat. 

This change converts the type into a string which prevents it from looping infinitely. I am not 100% sure if this is the best fix it but does fix it. 

Thanks!

rockerBOO committed:
Fix infinite loop for token format

myleshyson commented:
thanks for reporting! `types.go` is actually a generated file, so the change would have to be in python. Can you make that update in python? You'll need to make the change in `enums.py`

rockerBOO committed:
Add string(*t) to enum.py

rockerBOO commented:
I do not know how to run the generator to test but I have updated that part and reverted types.go

*Edit*: Oh I see from the workflow. So we'll need to figure out what the type is (uint32 or string) but it's a little more involved for me to figure out. Might take a little longer to get that in there.

rockerBOO committed:
Update to simpler dereference

rockerBOO commented:
Ok I think this works now. I am newer to Go though so let me know if there is some other tests we could add to test the JSON marshaling.

myleshyson commented:
Ok this is merged in now and released at 1.0.1. Thanks again 👍
